docs/src/osdk/reference/manifest.md

```diff
@@ -32,58 +32,103 @@ Below, you will find a comprehensive version of
 the available configuration options in the manifest.
 
 ```toml
-kcmd_args = ["init=/bin/busybox", "path=/usr/local/bin"] # <1>
-init_args = ["sh", "-l"] # <2>
-initramfs="./build/initramfs.cpio.gz" # <3>
-
-[boot]
-loader = "grub" # <4>
-protocol = "multiboot2" # <5>
-grub-mkrescue = "/usr/bin/grub-mkrescue" # <6>
-ovmf = "/usr/bin/ovmf" # <7>
-
-[qemu]
-path = "/usr/bin/qemu-system-x86_64" # <8>
-machine = "q35" # <9>
-args = [ # <10>
-    "-enable-kvm",
-    "-m 2G", 
-    "-device virtio-keyboard-pci,disable-legacy=on,disable-modern=off"
-] 
+[project] # <1>
+type = "kernel"
+
+[run] # <2>
+kcmd_args = ["init=/bin/busybox", "path=/usr/local/bin"] # <3>
+init_args = ["sh", "-l"] # <4>
+initramfs="./build/initramfs.cpio.gz" # <5>
+bootloader = "grub" # <6>
+boot_protocol = "multiboot2" # <7>
+grub-mkrescue = "/usr/bin/grub-mkrescue" # <8>
+ovmf = "/usr/bin/ovmf" # <9>
+qemu_path = "/usr/bin/qemu-system-x86_64" # <10>
+qemu_args = [ # <11>
+    "-machine q35,kernel-irqchip=split",
+    "-cpu Icelake-Server,+x2apic",
+    "-m 2G",
+]
+
+[test] # <2>
+bootloader = "qemu"
+qemu_args = [ # <10>
+    "-machine q35,kernel-irqchip=split",
+    "-cpu Icelake-Server,+x2apic",
+    "-m 2G",
+]
+
+['cfg(arch="x86_64", schema=microvm)'.run] # <12>
```

Comment:
Well, `schema` seems to in a different level from the `cfg`. I think we should provide it a more significant place.

How do you like the following style:
```toml
[schema."<name>".<profile>.<cfg>]
```
`name`, `profile` and `cfg` can all be ommitted. If name is ommitted, the schema is the default schema. `profile` can be `run` or `test`, in our cases. 

For example
```toml
[schema] # Default schema
[schema."microvm"] # For microvm
[schema."tdx"] # For TDX
[schema."microvm".test."cfg(arch = "x86_64")"] # For microvm, test, x86_64 target only.
```

There are two advantages:
1. `name`, `profile` and `cfg` can be combined in any ways, so they can provide most flexible configurations. 
2. This hierarchical structure is advantageous for inheriting parameters and avoiding duplication. For example, `schema.run.cfg(arch = "x86_64")` can inherit qemu arguments from `schema.run`, if needed.

Reply:
Sounds good.